aten/src/THC/generic/THCTensorMathReduce.cu

```diff
@@ -302,6 +302,7 @@ accreal THCTensor_(meanall)(THCState *state, THCTensor *self)
 
 scalar_t THCTensor_(minall)(THCState *state, THCTensor *self) {
   THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, self));
+  THArgCheck(THTensor_nDimensionLegacyAll(self) > 0, 1, "tensor must have one dimension");
```

Comment:
This is good because it makes the CPU and CUDA error messages match up.  Let me know if you want me to merge it as (can you fix the merge conflict?) or alternatively, I have a suggestion for improvement here.

This isn't quite the right error message (even in the CPU case which you didn't touch), and it's totally not clear why because this is legacy code -- sorry about that!

But basically, tensors with 0 elements, used to have 0-dimensions, and `THTensor_nDimensionLegacyAll` is properly checking that, but:
1) we shouldn't be using that API in any new code, we should just check that `self->numel() > 0`
2) as above, the error message isn't quite correct, because "must have one dimension" is referring to legacy dimension before, but what we really want is a message about identity that matches the case where a dimension is specified, which is "cannot perform reduction function max on tensor with no elements because the operation does not have an identity".